src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,266 @@
+            {
+                var semanticModel = operation.SemanticModel;
+                if (semanticModel is null)
+                {
+                    return;
+                }
+
+                var symbolInfo = semanticModel.GetSymbolInfo(operation.Syntax, cancellationToken);
+                if (symbolInfo.Symbol is null)
+                {
+                    return;
+                }
+
+                await AddOperationAsync(operation, symbolInfo.Symbol, cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task VisitChildrenAsync(IOperation operation, CancellationToken cancellationToken)
+            {
+                foreach (var child in operation.Children)
+                {
+                    await VisitAsync(child, cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            private Task VisitAssignmentOperationAsync(IAssignmentOperation assignmentOperation, CancellationToken cancellationToken)
+                => VisitAsync(assignmentOperation.Value, cancellationToken);
+
+            private Task VisitObjectCreationAsync(IObjectCreationOperation objectCreationOperation, CancellationToken cancellationToken)
+                => TrackArgumentsAsync(objectCreationOperation.Arguments, cancellationToken);
+
+            private async Task VisitInvocationAsync(IInvocationOperation invocationOperation, CancellationToken cancellationToken)
+            {
+                await AddOperationAsync(invocationOperation, invocationOperation.TargetMethod, cancellationToken).ConfigureAwait(false);
+                await TrackArgumentsAsync(invocationOperation.Arguments, cancellationToken).ConfigureAwait(false);
+            }
+
+            private Task VisitReferenceAsync(IOperation operation, CancellationToken cancellationToken)
+            {
+                Debug.Assert(operation is
+                    ILocalReferenceOperation or
+                    IParameterReferenceOperation or
+                    IFieldReferenceOperation or
+                    IPropertyReferenceOperation);
+
+                if (IsContainedIn<IArgumentOperation>(operation, out var argumentOperation) && argumentOperation.Parameter is not null)
+                {
+                    if (argumentOperation.Parameter.IsRefOrOut())
+                    {
+                        // Always add ref or out parameters to track as assignments since the values count as 
+                        // assignments across method calls for the purposes of value tracking.
+                        return AddOperationAsync(operation, argumentOperation.Parameter, cancellationToken);
+                    }
+
+                    // If the parameter is not a ref or out param, track the reference assignments that count
+                    // as input to the argument being passed to the method.
+                    return AddReference(operation, cancellationToken);
```

Comment:
i'm not sure if this is right or not.  if you have

```
Foo(x => x + p)
```

what should happen here?  `p` will be  parameter, and willb e in teh argument.  But should it count as necessarily flowing in, as it's just a capture in the lambda?  i honestly don't know :)

Reply:
I think it _should_ count similar to a non ref/out parameter to the lambda operation. Need to think more on it though... I haven't added any lambda tests yet...